docs/source/traitsui_user_manual/testing/discussions/working_of_extensions.rst

```diff
@@ -0,0 +1,40 @@
+.. _testing-how-extension-works:
+
+How extension works
+===================
+
+|UITester| supports testing TraitsUI editors with various user interaction
+logic. However it is possible that projects' test code may require
+additional logic that is not supported by |UITester| by default. Furthermore,
+some projects may implement and maintain their own custom UI editors. Those
+custom UI editors are also by default not supported by |UITester|.
+
+The API allows extension such that
+
+* projects can test TraitsUI editors with user interactions that do not
+  come supported by default.
+* projects can reuse the testing API for testing custom editors.
+
+As described in :ref:`testing-add-new-interaction` and
+:ref:`testing-add-new-location`, the testing API can be extended by providing
+one or many registry objects. Internally, |UITester| maintains a list of
+registries ordered in decreasing priority. For example, if you provide multiple
+registries::
+
+    tester = UITester(registries=[custom_registry, another_registry])
+
+Interaction and location support registered in the first registry will
+supersede that of the second (if such implementation exists).
+
+This list of registries is passed on to all |UIWrapper| created from the
+|UITester|. When |UIWrapper.perform| or |UIWrapper.inspect| is called, the
+first registry that can support the given target and interaction will be used.
+Likewise, |UIWrapper.locate| follows the same rule for the given target and
+location. See :ref:`testing-target-interaction-location` for an explanation on
+these three types of objects.
+
+This is how |UITester| supports testing on TraitsUI editor: It always extends
+the given list of registries with more built-in registries that know how to
```

Comment:
Is this strictly speaking correct?  I believe for TraitsUI editor implementations provided by default they are all provided by the single default registry.  So wee are not extending the list of registries for each editor we support, we are adding to the things we register on the default registry.  Although both would work fine

Reply:
I was future-proofing this sentence. The UITester can add more registries than just one. I did not think it would be interpreted as one-registry-per-editor though.

(Motivation, for https://github.com/enthought/traitsui/issues/1418, I was thinking the `_get_default_registry` function will be changed to return a list, rather than just one registry.)

Any suggestion to reword this? Otherwise I might be keeping this as is...

Reply:
maybe just remove the word "always"? But given the potential change coming from addressing 1418 I am perfectly happy to leave this as it is here.

Reply:
Removing 'always' sounds good to me!